File: content-solutions/dp/coin-ways.java
class Solution {
    public int solution(int[] coins, int amount) {
        final long MOD = 1_000_000_007L;
        long[] ways = new long[amount + 1];
        ways[0] = 1;
        for (int c : coins) {
            for (int a = c; a <= amount; a++) ways[a] = (ways[a] + ways[a - c]) % MOD;
        }
        return (int) ways[amount];
    }
}

File: content-solutions/dp/coin-ways.py
def solution(coins, amount):
    MOD = 1_000_000_007
    ways = [0] * (amount + 1)
    ways[0] = 1
    for c in coins:
        for a in range(c, amount + 1):
            ways[a] = (ways[a] + ways[a - c]) % MOD
    return ways[amount]
